kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -0,0 +1,486 @@
+    if (first > last) {
+      throw std::invalid_argument{"Invalid block range"};
+    }
+    auto raw_blocks = std::vector<std::pair<std::string, std::string>>{};
+    for (auto i = first; i <= last; ++i) {
+      const auto raw_block = adapter.getRawBlock(i);
+      if (!raw_block) {
+        throw NotFoundException{"Couldn't find a block by ID = " + std::to_string(i)};
+      }
+      raw_blocks.emplace_back("rawBlock" + std::to_string(i),
+                              concordUtils::vectorToHex(categorization::RawBlock::serialize(*raw_block)));
+    }
+    return toJson(raw_blocks);
+  }
+};
+
+struct GetBlockInfo {
+  bool read_only = true;
+  std::string description() const {
+    return "getBlockInfo BLOCK-ID\n"
+           "  Returns information about the requested block (excluding its key-values).";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.empty()) {
+      throw std::invalid_argument{"Missing BLOCK-ID argument"};
+    }
+    auto blockId = toBlockId(args.values.front());
+    const auto parent_digest = adapter.parentDigest(blockId);
+    const auto updates = adapter.getBlockUpdates(blockId);
+    if (!updates) {
+      throw NotFoundException{"Couldn't find a block by ID = " + std::to_string(blockId)};
+    }
+    auto catUpdates = updates->categoryUpdates();
+    size_t keyValueCount = 0;
+    for (auto const &catkv : catUpdates.kv) {
+      keyValueCount += std::visit([](auto &&arg) { return arg.kv.size(); }, catkv.second);
+    }
+
+    return toJson(std::map<std::string, std::string>{
+        std::make_pair(
+            "parentBlockDigest",
+            parent_digest ? concordUtils::bufferToHex(parent_digest->data(), parent_digest->size()) : "n/a"s),
+        std::make_pair("keyValueCount", std::to_string(keyValueCount)),
+        std::make_pair("categoriesCount", std::to_string(catUpdates.kv.size()))});
+  }
+};
+
+inline std::map<std::string, std::string> getKVStr(
+    const std::variant<BlockMerkleInput, VersionedInput, ImmutableInput> &val) {
+  std::map<std::string, std::string> kvout;
+  std::visit(
+      [&kvout](auto &&arg) {
+        for (auto const &c : arg.kv) {
+          using T = std::decay_t<decltype(arg)>;
+          if constexpr (std::is_same_v<T, BlockMerkleInput>) {
+            kvout[concordUtils::bufferToHex(c.first.data(), c.first.size())] =
+                concordUtils::bufferToHex(c.second.data(), c.second.size());
+          } else if constexpr (std::is_same_v<T, VersionedInput>) {
+            kvout[concordUtils::bufferToHex(c.first.data(), c.first.size())] =
+                concordUtils::bufferToHex(c.second.data.data(), c.second.data.size());
+          } else if constexpr (std::is_same_v<T, ImmutableInput>) {
+            kvout[concordUtils::bufferToHex(c.first.data(), c.first.size())] =
+                concordUtils::bufferToHex(c.second.data.data(), c.second.data.size());
+          }
```

Comment:
I guess these two `else if` statements can be combined into one as they are having the same "interface".